### PR TITLE
[WIP] RSpec conversion

### PR DESCRIPTION
**Pull request by smashwilson**
As per #7, I'm translating the exist scaffolding tests to RSpec and fleshing them out a little while I go. You can run 'em with `bundle exec rake spec`.

Aesthetics-wise, I'm fairly ambivalent to RSpec over Test::Unit; some claim that `expect(something).to eq(something_else)` is easier to work with because it reads more like English, but personally I'd never thought that `assert_equals something, something_else` was particularly unclear. (Actually, although I'm converting them as I go to show the changes, apparently [you can use Rails-style assertions freely](https://www.relishapp.com/rspec/rspec-rails/v/2-14/docs/controller-specs) in specs.) Mainly, I like RSpec because it comes out of the box with a few nice utility features:
### Pending tests

You can start by literally writing out a TODO list:

``` ruby
describe "a thing" do
  it "does something"
  it "does something else"
  it "does a third thing"
end
```

... and you'll get a nice list of 'pending' specs in your output, distinct from and less angry than your failing ones.
### Expectations and Stubbing

You can easily mock out dependent objects, or set up "expectations" on external calls that should be made, including the arguments that should be present. This is especially useful for things like Sidekiq workers!

``` ruby
# Don't actually call the method, just make it look like did.
allow(ExternalObject).to receive(:costly_method).and_return(123)

# Fail the test if BuildWorker.perform_async doesn't get called by the end!
expect(BuildWorker).to receive(:perform_async).with(build.id).and_return(true)
```

### CODE REVIEW

**Comment by smashwilson** [comment]
Oh hey, apparently ActiveSupport already has a [`pending` implementation](http://stackoverflow.com/questions/5467481/how-to-use-the-pending-module-in-activesupport). I was unaware!

File: Gemfile [hunk 1]
```diff
@@ -16,3 +16,8 @@ gem 'dotenv-rails'
 gem 'foreman'
 gem 'sidekiq'
 gem 'sinatra', '>= 1.3.0', :require => nil
+
+group :development do
```
Comment (cobyism) [inline Gemfile:20]: Does this need to include the `:test` group too?

**Comment by cobyism** [comment]
Going to go ahead and merge this in. Iteration on anything else will be easier if we at least have some basic tests to add to :grin: Thanks a bunch for this!

File: Gemfile (hunk 1 above)
Comment (smashwilson) [inline Gemfile:20]: Yes, yes it should. :stuck_out_tongue: Want me to just patch it in on master? If you haven't gotten to it yourself, yet, that is.

Comment (cobyism) [inline Gemfile:20]: Yeah, just patch it on `master`. :+1:

**Pull request merged**